sdk/security_keyvault/src/secrets/operations/set_secret.rs

```diff
@@ -21,7 +21,8 @@ impl SetSecretBuilder {
 
             let body = Some(Value::Object(request_body).to_string().into());
 
-            let headers = Headers::new();
+            let mut headers = Headers::new();
+            headers.insert(CONTENT_TYPE, "application/json");
```

Comment:
This appears to be a larger problem than just SetSecretBuilder.  Setting the content type was mistakenly dropped in https://github.com/Azure/azure-sdk-for-rust/pull/962.

Instead of updating just SetSecretBuilder, can you instead update `finalize_request`?

Perhaps something like this:
```diff
--- a/sdk/security_keyvault/src/clients/keyvault_client.rs
+++ b/sdk/security_keyvault/src/clients/keyvault_client.rs
@@ -4,7 +4,8 @@ use azure_core::{
     date,
     error::{Error, ErrorKind},
     headers::*,
-    Body, Context, Method, Pipeline, Request, Response,
+    prelude::*,
+    Body, Method, Pipeline, Request, Response,
 };
 use const_format::formatcp;
 use std::sync::Arc;
@@ -84,6 +85,9 @@ impl KeyvaultClient {
         request.insert_header(MS_DATE, time);

         if let Some(request_body) = request_body {
+            if request.headers().get_optional_str(&CONTENT_TYPE).is_none() {
+                request.insert_headers(&ContentType::APPLICATION_JSON);
+            }
             request.insert_header(CONTENT_LENGTH, request_body.len().to_string());
             request.set_body(request_body);
         } else {
```

Reply:
Note, if you'd prefer, let me know and I can make the broader change.

Thanks!

Reply:
I can do it, I was just thinking that the other operations (get, delete, get_versions) are working without the `Content-Type` header, hence putting it into the common `finalize_request` method seemed incorrect to me.

Reply:
but your suggestion make sense that for request with body it should be added, in case in the future there are other operation which need to set the header. I will change it according to your suggestion.

Reply:
get & delete do not set request bodies, which makes this less of an issue.